sdk/cosmos/src/requests/replace_document_builder.rs

```diff
@@ -62,12 +45,14 @@ where
         if_modified_since: &'b DateTime<Utc> => Some(IfModifiedSince::new(if_modified_since)),
         allow_tentative_writes: TenativeWritesAllowance,
         indexing_directive: IndexingDirective,
+        partition_keys: PartitionKeys => Some(partition_keys),
     }
 }
 
-impl<'a, 'b> ReplaceDocumentBuilder<'a, 'b, Yes, Yes> {
-    pub async fn execute_with_document<T>(
+impl<'a, 'b> ReplaceDocumentBuilder<'a, 'b> {
+    pub async fn execute<T>(
         &self,
+        document_id: &str,
```

Comment:
Should this be a parameter to the `replace_document` function instead? This is a configuration parameter rather than data